gamestonk_terminal/stocks/options/yfinance_view.py

```diff
@@ -662,3 +666,236 @@ def risk_neutral_vals(
+    """
+    letters = [
+        "A",
+        "B",
+        "C",
+        "D",
+        "E",
+        "F",
+        "G",
+        "H",
+        "I",
+        "J",
+        "K",
+        "L",
+        "M",
+    ]
+    letters += [
+        "N",
+        "O",
+        "P",
+        "Q",
+        "R",
+        "S",
+        "T",
+        "U",
+        "V",
+        "W",
+        "X",
+        "Y",
+        "Z",
+    ]
+    opts = (
+        [f"{x}" for x in letters]
+        + [f"{x}{y}" for x in letters for y in letters]
+        + [f"{x}{y}{z}" for x in letters for y in letters for z in letters]
+    )
+    wb = Workbook()
+    ws = wb.active
+
+    ws["A1"] = "U"
+    ws["B1"] = up
+    ws["A2"] = "D"
+    ws["B2"] = 1 / up
+    ws["D1"] = "q"
+    ws["E1"] = prob_up
+    ws["D2"] = "1-q"
+    ws["E2"] = 1 - prob_up
+    ws["D3"] = "Discount"
+    ws["E3"] = discount
+    ws["A4"] = "Binomial Tree for Underlying Values"
+    for i, _ in enumerate(und_vals):
+        for j, _ in enumerate(und_vals[i]):
+            ws[f"{opts[i]}{j+5}"] = und_vals[i][j]
+
+    ws[f"A{days+7}"] = "Binomial Tree for Option Values"
+    for i, _ in enumerate(opt_vals):
+        for j, _ in enumerate(opt_vals[i]):
+            ws[f"{opts[i]}{j+8+days}"] = opt_vals[i][j]
+
+    trypath = os.path.join(
+        os.path.abspath(os.path.join(os.path.dirname(__file__), "..", "..", "..")),
+        "exports",
+        "stocks",
+        "options",
+        f"{ticker} {datetime.now()}.xlsx",
+    )
+    wb.save(trypath)
+    print(f"Analysis ran for {ticker}\nPlease look in {trypath} for the file.\n")
+
+
+def show_binom(
+    ticker: str,
+    expiration: str,
+    strike: float,
+    put: bool,
+    europe: bool,
+    export: bool,
+    plot: bool,
+    vol: float,
+) -> None:
+    """Get binomial pricing for option
+
+    Parameters
+    ----------
+    ticker : str
+        The ticker of the option's underlying asset
+    expiration : str
+        The expiration for the option
+    strike : float
+        The strike price for the option
+    put : bool
+        Value a put instead of a call
+    europe : bool
+        Value a European option instead of an American option
+    export : bool
+        Export the options data to an excel spreadsheet
+    plot : bool
+        Show a graph of expected ending prices
+    vol : float
+        The annualized volatility for the underlying asset
+    """
+    # Base variables to calculate values
+    info = yfinance_model.get_info(ticker)
+    price = info["regularMarketPrice"]
+    if vol is None:
+        closings = yfinance_model.get_closing(ticker)
+        vol = (closings / closings.shift()).std() * (252 ** 0.5)
+    div_yield = (
+        info["trailingAnnualDividendYield"]
+        if info["trailingAnnualDividendYield"] is not None
+        else 0
+    )
+    delta_t = 1 / 252
+    rf = get_rf()
+    exp_date = datetime.strptime(expiration, "%Y-%m-%d").date()
+    today = date.today()
+    days = (exp_date - today).days
```

Comment:
Question (something I ran into when looking at black Scholes pricing):  If you look at 0dte options, then days will be 0.  This causes some things to diverge (maybe not here).  So should this actually be in hours?  That way you have decimal days til expiry?

Note that you don't need to change it, just a question I am not sure about.

Reply:
I do not have the data but I would venture most options on the date of expiration trade for pretty close to expiration because of they are expiring so soon. This means even with no additional branches being made we will still be decently close to the real value. I think focusing on a better measure of volatility will be much better for the tools overall predictive value.